tf-modules/archive/iam.tf

```diff
@@ -128,6 +128,13 @@ data "aws_iam_policy_document" "lambda_api_gateway_policy" {
     resources = ["arn:aws:sqs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:${var.prefix}-*"]
   }
 
+  statement {
+    actions = [
+      "states:StartExecution",
+    ]
+    resources = ["arn:aws:states:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:stateMachine:${var.prefix}-*"]
+  }
```

Comment:
Huh. I'm kinda surprised we needed a new policy here for this. I might just need a refresh on how our tf modules are organized but the `archive` module has bulk operations, which makes me curious. 

I see we have a `states:StartExecution` action in `lambda_processing_policy` on the cumulus module. Would that not cover this API endpoint? I would have thought so because tf-modules/cumulus/archive.tf inherits from tf-modules/archive...?

Reply:
It's a good question - both of the API lambdas are explicitly set to this role (e.g. the `role` key in `resource "aws_lambda_function" "api"`).  

The policy is applied to the role here

```
resource "aws_iam_role_policy" "lambda_api_gateway" {
  name   = "${var.prefix}_lambda_api_gateway_policy"
  role   = aws_iam_role.lambda_api_gateway.id
  policy = data.aws_iam_policy_document.lambda_api_gateway_policy.json
}
```

The lambda processing role is passed in to the archive module for things like bootstrap/bulk operation but *not* the API itself.

Reply:
Ah. Thanks for the explanation!